### ISSUE TITLE
fatal error: lgc.h: No such file or directory

### ISSUE THREAD

**Issue by tomimaki** [scp-fs2open/fs2open.github.com#3650]
Seems I can no longer use lua installed in system. It is liblua5.1-0-dev on Debian.
Nevertheless compilation works if I use lua from source (i.e. if cmake cant find lua5.1).
```
/home/tomi/fs2open.github.com/freespace2/freespace.cpp:194:11: fatal error: lgc.h: No such file or directory
  194 |  #include "lgc.h"
      |           ^~~~~~~
compilation terminated.
```
It was added in #3635

**Comment by Goober5000** [comment]
I suppose we will need @asarium or someone else familiar with the build system to chime in.  The file lgc.h is included as part of the FSO source code, but I wouldn't know what configurations are necessary to allow a different lgc.h to be used.  I imagine the #include directories would need to be set via CMake.

**Comment by asarium** [comment]
`lgc.h` is not part of the public API of lua and as such may or may not be in an include path somewhere. The code added in #3635 needs to use `lua_gc` from the public API to fix this issue.

**Comment by Goober5000** [comment]
This should be fixed in #3651.
